components/tidb_query/src/expr_util/time.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Common utility implementations for time expression
+
+pub fn period_to_month(period: u64) -> u64 {
+    if period == 0 {
+        return 0;
+    }
+    let (year, month) = (period / 100, period % 100);
+    if year < 70 {
+        (year + 2000) * 12 + month - 1
+    } else if year < 100 {
+        (year + 1900) * 12 + month - 1
+    } else {
+        year * 12 + month - 1
+    }
+}
+
```

Comment:
what if the `period` is pretty large, like: `1234567891011`.
in mysql, it reports:
```sql
+------------------------------+
| period_add(1234567891011, 0) |
+------------------------------+
| 0                            |
+------------------------------+

```

Reply:
What version of mysql did you get? Tested both in version **5.7.22** and **5.6.20-log** , i get the same result as the push-down functions

Reply:
`mysql  Ver 15.1 Distrib 10.4.10-MariaDB, for Linux (x86_64) using readline 5.1`